### PR TITLE
Story about range's copyFrom method

### PR DESCRIPTION
**Pull request by AlexJerabek**

### CODE REVIEW

**Comment by o365devx** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-ranges.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/excel/excel-add-ins-ranges.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C17%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808172049281813-314%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=uIvrGbScAY9XW1QoOlCE%2F%2F5g6CxSRLv9iVFD95g533I%3D&st=2018-08-17T20%3A45%3A32Z&se=2018-09-17T20%3A50%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-ranges.md [hunk 1]
```diff
@@ -531,6 +531,44 @@ Excel.run(function (context) {
...
+```js
+Excel.run(function (context) {
+    var sheet = context.workbook.worksheets.getItem("Sample");
+    // copy a range starting at a single cell destination
+    sheet.getRange("G1").copyFrom("A1:E1");
+    return context.sync();
+}).catch(errorHandlerFunction);
+```
+
+Range.copyFrom has three optional parameters, `copyType`, `skipBlanks`, and `transpose`.
```
[the remark this answers is not in the document]
Comment (AlexJerabek) [reply docs/excel/excel-add-ins-ranges.md]: @Rick-Kirkham I was wondering how much detail to go into for this doc. I had a more verbose version explaining all the parameters, but I realized I was just writing the reference docs. Is it worth going into detail here, or should I just link to the reference (which for now is the beta CDN d.ts)

**Comment by o365devx** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-ranges.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/excel/excel-add-ins-ranges.md) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-after.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-after.png) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C17%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808172211335967-314%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=nNrvIiIMgl6QB0iTecj12t2TWlgTLZjVgH8E6eKJRCI%3D&st=2018-08-17T22%3A07%3A38Z&se=2018-09-17T22%3A12%3A38Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-ranges.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/excel/excel-add-ins-ranges.md) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-after.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-after.png) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C17%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808172212560577-314%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=uL8y9PcN%2BfPB0IByyxyvKgW5tbr4YDkVfyeVijNojGw%3D&st=2018-08-17T22%3A09%3A10Z&se=2018-09-17T22%3A14%3A10Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-ranges.md [hunk 2]
```diff
@@ -531,6 +531,70 @@ Excel.run(function (context) {
 
 ![Data in Excel after format is set](../images/excel-ranges-format-numbers.png)
 
+## Copy and Paste
+
+> [!NOTE]
+> The copyFrom function is currently available only in public preview (beta). To use this feature, you must use the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+> If you are using TypeScript or your code editor uses TypeScript type definition files for Intellisense, use https://appsforoffice.microsoft.com/lib/beta/hosted/office.d.ts.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-ranges.md]: IntelliSense - caps on the S

File: docs/excel/excel-add-ins-ranges.md [hunk 3]
```diff
@@ -585,9 +585,13 @@ Excel.run(function (context) {
...
 *Before the preceeding function has been run.*
+
 ![Data in Excel before range’s copy method has been run.](../images/excel-range-copyfrom-skipblanks-before.png)
+
 *After the preceeding function has been run.*
-![Data in Excel after range’s copy method has been run.](../images/ excel-range-copyfrom-skipblanks-after.png)
+
+![Data in Excel after range’s copy method has been run.](../images/excel-range-copyfrom-skipblanks-after.png)
+
 `transpose` determines whether the data is algebraically transposed into the source location. A transposed range has its rows and columns switched; flipping the cells along the main diagonal.
```
Comment (mscharlock) [inline docs/excel/excel-add-ins-ranges.md]: Not sure you should use transpose in a definition of transpose?

**Comment by o365devx** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-ranges.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/excel/excel-add-ins-ranges.md) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-after.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-after.png) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C17%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808172311510737-314%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=yCml3cGEoIzX9%2BWhfnAtYsxSCyfFaptqPopPrCK9twg%3D&st=2018-08-17T23%3A08%3A14Z&se=2018-09-17T23%3A13%3A14Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-ranges.md (hunk 3 above)
Comment (AlexJerabek) [reply docs/excel/excel-add-ins-ranges.md]: I would argue I'm defining a parameter called "transpose" here and specifying what the specific usage of that word is.  There's [non-algebra definitions](https://www.merriam-webster.com/dictionary/transpose) too :-)

**Comment by o365devx** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-ranges.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/excel/excel-add-ins-ranges.md) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-after.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-after.png) | :white_check_mark:Succeeded | |
[docs/images/excel-range-copyfrom-skipblanks-before.png](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-CopyPaste/docs/images/excel-range-copyfrom-skipblanks-before.png) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C17%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808172334029647-314%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=Xwn1hvFF1zCDiev7J4QL1%2Flz0ZM2fZrruYv3eiiORHM%3D&st=2018-08-17T23%3A30%3A12Z&se=2018-09-17T23%3A35%3A12Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/excel-add-ins-ranges.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rick-Kirkham) [reply docs/excel/excel-add-ins-ranges.md]: @AlexJerabek We certainly don't need to repeat the whole ref documentation in the story. How much to say depends on how self-documenting the parameters are named. This is an unusual case, in which all 3 were puzzling. For example, "skipBlank" might have meant "don't overwrite black spaces in the target". And "transpose" was completely mysterious to me.

**Pull request merged**